### ISSUE TITLE
new rule - enforce (or ban) parenthesized tuples everywhere

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#13342]
[`incorrectly-parenthesized-tuple-in-subscript` (`RUF031`)](https://docs.astral.sh/ruff/rules/incorrectly-parenthesized-tuple-in-subscript/) allows you to enforce (or ban) them inside subscripts, but imo there should also be a rule to enforce it everywhere else as well:

```py
foo = 1, 2 # error: tuple should be wrapped in parentheses
```

mainly for code consistency reasons.

**Comment by MichaReiser** [comment]
I did a quick look and such a rule would be incompatible with the formatter when tuples are used in a `for` loop target because the formatte removes those parentheses ([playground](https://play.ruff.rs/0632c292-29de-40ae-8bb5-5c699696c688)).

**Comment by DetachHead** [comment]
perhaps an exception can be made when the tuple is used to unpack a value, eg:

```py
for a, b in (1, 2), (3, 4):
    pass
a, b = 1, 2
```
would get converted to:
```py
for a, b in ((1, 2), (3, 4)):
    pass
a, b = (1, 2)
```

**Comment by tdulcet** [comment]
I would support a rule that removes unnecessary parentheses, but either way, improving constancy in a code base would be great.
